rhizome/common/lib/util.rb

```diff
@@ -58,10 +58,16 @@ def sync_parent_dir(f)
 end
 
 def safe_write_to_file(filename, content)
+  safe_create_file_with_content(filename) do |f|
+    f.puts(content)
+  end
+end
+
+def safe_create_file_with_content(filename, &block)
```

Comment:
By just reading the function names, I'd expect `safe_create_file_with_content` to get `content` and `safe_write_to_file` to  get `block`. I guess you kept it this way to avoid changing existing references to `safe_write_to_file`.

One alternative you can do is having just one function, which can accept both content and block and its behavior would change based on whether block is passed or not. You can check if block is passed with `block_given?` helper. This is common pattern in ruby. We even have an example in our codebase at lib/clog.rb

Reply:
ok, if it's confusing I'll collapse it into a single function.